src/OrchardCore/OrchardCore.ContentManagement.Abstractions/ContentItemExtensions.cs

```diff
@@ -28,6 +28,16 @@ public static TPart GetOrCreate<TPart>(this ContentItem contentItem) where TPart
             return contentItem.GetOrCreate<TPart>(typeof(TPart).Name);
         }
 
+        /// <summary>
+        /// Removes a content part by its type.
+        /// </summary>
+        /// <param name="contentItem">The <see cref="ContentItem"/>.</param>
+        /// <typeparam name="TPart">The type of the content part.</typeparam>
+        public static void Remove<TPart>(this ContentItem contentItem) where TPart : ContentPart, new()
```

Comment:
Had to introduce a new `Remove` extension to drop it from the `Elements` and `Data` dictionaries.